deploy/rbd/helm/templates/provisioner-clusterrole.yaml

```diff
@@ -28,4 +28,19 @@ rules:
   - apiGroups: [""]
     resources: ["endpoints"]
     verbs: ["get", "create", "update"]
+  - apiGroups: ["snapshot.storage.k8s.io"]
+    resources: ["volumesnapshots"]
+    verbs: ["get", "list", "watch", "update"]
+  - apiGroups: [""]
+    resources: ["configmaps"]
+    verbs: ["get", "list", "create", "delete"]
+  - apiGroups: ["snapshot.storage.k8s.io"]
+    resources: ["volumesnapshotcontents"]
+    verbs: ["create", "get", "list", "watch", "update", "delete"]
+  - apiGroups: ["snapshot.storage.k8s.io"]
+    resources: ["volumesnapshotclasses"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["apiextensions.k8s.io"]
+    resources: ["customresourcedefinitions"]
+    verbs: ["create"]
```

Comment:
I was with you up to the CRD? why does the provisioner need the ability to create Custom Resource Definitions? Should we remove this from both sets of manifests?

Reply:
because snapshot controller is now part of the pod, it requires all the resources (including crd) to get snapshot objects.

Reply:
I don't follow. This rule only allows it to create a CRD. ie, register a new type of custom resource with kubernetes. not check for existing ones or create/get your own custom resources.